core/common/privdata/collection.go

```diff
@@ -29,7 +29,7 @@ type Collection interface {
 
 	// MemberOrgs returns the collection's members as MSP IDs. This serves as
 	// a human-readable way of quickly identifying who is part of a collection.
-	MemberOrgs() []string
```

Comment:
We can still keep it as `[]string` as the goal is to minimize changes made to release 1.4. 
We have two other options: 

1. we can convert the map to `[]string` at https://github.com/hyperledger/fabric/blob/1007a5f21fb6b2821535a4e23f2acfe9924122af/core/common/privdata/simplecollection.go#L89
2. we can return `[]string` from `getMemberOrgs` instead of a map. We can do a linear search on the `[]string` to find the match. Given that we use this function only during the chaincode upgrade, no optimization (usage of the map) is required IMO. If optimization is required (though it is not), we in fact need to check for the match after each deserialization of identity rather than collecting all identities at once and then do the matching. 

I think we can go with option (2).

Reply:
I believe that was what it was before, I originally changed to a map via an earlier suggestion. I do see that the extent of the change is a little too broad for 1.4 though so I'll probably just do the map in the 2.0/master branch fix.